op-devstack/dsl/l2_cl.go

```diff
@@ -117,37 +151,68 @@ func (cl *L2CLNode) NotAdvanced(lvl types.SafetyLevel, attempts int) CheckFunc {
 // Composable with other lambdas to wait in parallel
 func (cl *L2CLNode) Reached(lvl types.SafetyLevel, target uint64, attempts int) CheckFunc {
 	return func() error {
-		cl.log.Info("expecting chain to reach", "id", cl.inner.ID(), "chain", cl.chainID, "label", lvl, "target", target)
+		cl.log.Info("expecting chain to reach", "id", cl.inner.ID(), "chain", cl.ChainID(), "label", lvl, "target", target)
 		return retry.Do0(cl.ctx, attempts, &retry.FixedStrategy{Dur: 2 * time.Second},
 			func() error {
 				head := cl.HeadBlockRef(lvl)
 				if head.Number >= target {
-					cl.log.Info("chain advanced", "id", cl.inner.ID(), "chain", cl.chainID, "label", lvl, "target", target)
+					cl.log.Info("chain advanced", "id", cl.inner.ID(), "chain", cl.ChainID(), "label", lvl, "target", target)
 					return nil
 				}
-				cl.log.Info("Chain sync status", "id", cl.inner.ID(), "chain", cl.chainID, "label", lvl, "target", target, "current", head.Number)
+				cl.log.Info("Chain sync status", "id", cl.inner.ID(), "chain", cl.ChainID(), "label", lvl, "target", target, "current", head.Number)
 				return fmt.Errorf("expected head to advance: %s", lvl)
 			})
 	}
 }
 
+// ReachedRef is same as Reached, but has an additional check to ensure that the block referenced is not reorged
+// Composable with other lambdas to wait in parallel
+func (cl *L2CLNode) ReachedRef(lvl types.SafetyLevel, target eth.BlockID, attempts int) CheckFunc {
+	return func() error {
+		cl.log.Info("expecting chain to reach block ref", "id", cl.inner.ID(), "chain", cl.ChainID(), "label", lvl, "target", target)
+		err := retry.Do0(cl.ctx, attempts, &retry.FixedStrategy{Dur: 2 * time.Second},
```

Comment:
Suggestion: Can we call `Reached` for this?
```go
err := cl.Reached(lvl, target.Number, attempts)()
```

Reply:
No, because we care that a specific block is not reorged. `target.Number` will be reached, the question is which block precisely.

Reply:
What I meant was after checking the target block number is reached using `err := cl.Reached(lvl, target.Number, attempts)()`, we may check the hash.
So
```go
                if err := cl.Reached(lvl, target.Number, attempts)(); err != nil { return err }
		ethclient := cl.inner.ELs()[0].EthClient()
		result, err := ethclient.BlockRefByNumber(cl.ctx, target.Number)
		if err != nil {
			return err
		}
		if result.Hash != target.Hash {
			return fmt.Errorf("expected block ref to be the same as target %s, got but %s", target.Hash, result.Hash)
		}
		return nil

```